src/main/java/net/minecraftforge/client/gui/overlay/ForgeLayeredDraw.java

```diff
@@ -0,0 +1,241 @@
+    public static final ResourceLocation    TITLE_OVERLAY = ResourceLocation.withDefaultNamespace("title");
+    public static final ResourceLocation     CHAT_OVERLAY = ResourceLocation.withDefaultNamespace("chat_overlay");
+    public static final ResourceLocation         TAB_LIST = ResourceLocation.withDefaultNamespace("tab_list");
+    public static final ResourceLocation SUBTITLE_OVERLAY = ResourceLocation.withDefaultNamespace("subtitle");
+
+    public static final ResourceLocation    COMBINE_PHASE = ResourceLocation.withDefaultNamespace("combine_phase");
+    public static final ResourceLocation    SLEEP_OVERLAY = ResourceLocation.withDefaultNamespace("sleep_overlay");
+
+    /**
+     * Creates a stack of pre-named layers for vanilla.
+     * Not intended for modder use. Use {@linkplain ForgeLayeredDraw(ResourceLocation, Boolean, String...)} instead.
+     * @param layers Head of varargs marks the phase, the remaining args mark pre-named layers.
+     */
+    @ApiStatus.Internal
+    public ForgeLayeredDraw(ResourceLocation rl, String... layers) {
+        phase = rl;
+        mayEdit = true;
+        for (String layer : layers) {
+            order.add(ResourceLocation.withDefaultNamespace(layer));
+        }
+    }
+
+    /**
+     * Creates an empty draw list. Add entries with {@linkplain ForgeLayeredDraw#add(ResourceLocation, Layer)}
+     * @param rl marker for which phase this is.
+     * @param mayEdit whether other mods are allowed to insert layers into this list.
+     */
+    public ForgeLayeredDraw(ResourceLocation rl, boolean mayEdit) {
+        phase = rl;
+        this.mayEdit = mayEdit;
+    }
+
+    /**
+     * Creates a list of pre-defined names for layers to be added via {@linkplain ForgeLayeredDraw#add(Layer)}
+     * @param rl Used as the marker for what specific LayeredDraw phase this is.
+     * @param mayEdit Optionally fire ModifyOverlayLayers for this modded stack.
+     * @param layers List of pre-named layers.
+     */
+    public ForgeLayeredDraw(ResourceLocation rl, boolean mayEdit, String... layers) {
+        phase = rl;
+        this.mayEdit = mayEdit;
+        for (String layer : layers) {
+            order.add(ResourceLocation.fromNamespaceAndPath(rl.getPath(), layer));
+        }
+    }
+
+    /**
+     * Adds a layer with an already known name provided by {@linkplain ForgeLayeredDraw(ResourceLocation, Boolean, String...)}
+     * The layer will be rendered last (on top) of already added layers.
+     * Use {@linkplain ForgeLayeredDraw#add(ResourceLocation, Layer)} for adding individual layers.
+     * @param layer layer render code, see {@linkplain Layer} and example usages in {@linkplain net.minecraft.client.gui.Gui}
+     * @return this
+     */
+    @Override
+    public LayeredDraw add(Layer layer) {
+        if (count > order.size()) {
+            throw new IllegalStateException("Ran out of pre-named layers to use during " + phase);
```

Comment:
To keep computability with other systems you should use `"custom_"  + idx++` when you run out of expected ids.